Composer/plugins/azurePublish/src/node/azureResourceManager/azureResourceManager.ts

```diff
@@ -203,6 +205,214 @@ export class AzureResourceMananger {
     }
   }
 
+  /**
+   * QnA Resource depends on serveral components, including appinights and web config
+   * @param config 
+   */
+  public async deployQnAReource(config: QnAResourceConfig): Promise<{ endpoint: string; subscriptionKey: string }> {
+    try {
+      this.logger({
+        status: BotProjectDeployLoggerType.PROVISION_INFO,
+        message: 'Deploying QnA Resource ...',
+      });
+
+      // initialize the name
+      const qnaMakerSearchName = `${config.accountName}-search`.toLowerCase().replace('_', '');
+      const qnaMakerWebAppName = `${config.accountName}-qnahost`.toLowerCase().replace('_', '');
+      const qnaMakerServiceName = `${config.accountName}-qna`;
+      // deploy search service
+      const searchManagementClient = new SearchManagementClient(this.creds, this.subscriptionId);
+      const searchServiceDeployResult = await searchManagementClient.services.createOrUpdate(config.resourceGroupName, qnaMakerSearchName, {
+        location: config.location,
+        sku: {
+          name: 'standard'
+        },
+        replicaCount: 1,
+        partitionCount: 1,
+        hostingMode: 'default'
+      });
+
+      if (searchServiceDeployResult._response.status >= 300) {
+        this.logger({
+          status: BotProjectDeployLoggerType.PROVISION_ERROR,
+          message: searchServiceDeployResult._response.bodyAsText,
+        });
+        throw new Error(searchServiceDeployResult._response.bodyAsText);
+      }
+
+      // deploy websites
+      // Create new Service Plan or update the exisiting service plan created before
+      const webSiteManagementClient = new WebSiteManagementClient(this.creds, this.subscriptionId);
```

Comment:
If this has already been done as part of the hosting provisioning, do we need to do it again? 

Unless this is strictly necessary, I would propose we store and then re-use the necessary values from that part of the process.

Reply:
> If this has already been done as part of the hosting provisioning, do we need to do it again?
> 
> Unless this is strictly necessary, I would propose we store and then re-use the necessary values from that part of the process.

Yes, I think this website instance is a part of the QnA Maker, this resource is the same as [the resource](https://github.com/microsoft/BotFramework-Composer/blob/0f8efe29dbdf7d42a758b3bfe04b0bffef7a25e7/Composer/plugins/samples/assets/shared/scripts/DeploymentTemplates/qna-template.json#L165) in the qna arm template file.